classes/app.rb

```diff
@@ -0,0 +1,164 @@
+    until list_options
+      input = gets.chomp
+      if input == '7'
+        puts 'Thanks for using School Library App (^_^)'
+        break
+      end
+      option input
+    end
+  end
+
+  def list_options
+    puts('-------------------------')
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals of given person id'
+    puts '7 - Quit'
+    puts('-------------------------')
+    print 'Enter number of the operation: '
+  end
+
+  def option(input)
+    case input
+    when '1'
+      list_books
+    when '2'
+      list_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_rentals
+    else
+      'Enter digit from 1 to 7'
+    end
+  end
+
+  def list_books
+    if @books.empty?
+      puts 'No Books Available'
+    else
+      @books.each_with_index { |book, idx| puts "#{idx})  Book: #{book.title}, Author: #{book.author}" }
+    end
+    back_to_menu
+  end
+
+  def create_book
+    print 'Enter Book Title: '
+    title = gets.chomp
+    print 'Enter Book Author: '
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books.push(book)
+    puts 'Book Successfully Created'
+    back_to_menu
+  end
+
+  def create_person
+    puts '1 - Create Student  2 - Create Teacher'
+    select_option
+    person = gets.chomp
+    print 'Insert age: '
+    age = gets.chomp
+    print 'Insert name: '
+    name = gets.chomp
+    person_condition(person, age, name)
+    puts 'Person Created Successfully'
+    back_to_menu
+  end
+
+  def person_condition(person, age, name)
+    case person
+    when '1'
+      print 'Has parents permission [Y/N]: '
+      permit = gets.chomp.downcase
+      parent_permission = permit != 'n'
+      @people.push(Student.new(age, parent_permission, name))
+    when '2'
+      print 'Insert specialization: '
+      specialization = gets.chomp
+      teacher = Teacher.new(specialization, age, name)
+      @people.push(teacher)
+    end
+  end
+
+  def list_people
+    puts 'No person available' if @people.empty?
+    @people.each { |person| puts "[#{person.type}] Name: #{person.name} Age: #{person.age}" }
+    back_to_menu
+  end
+
+  def create_rental
+    if @books.empty?
+      puts 'No Books Available'
+    elsif @people.empty?
+      puts 'No Person Available'
+    else
+      puts 'Select a book from the following list by number'
+      @books.each_with_index { |book, index| puts "#{index}) Book Title: #{book.title}, Author: #{book.author}" }
+      rental_book = gets.chomp.to_i
+      puts 'Select a person from the following list by number (not id)'
+      @people.each_with_index do |person, index|
+        puts "#{index}) #{[person.type]} Name: #{person.name} Age: #{person.age} Id: #{person.id}"
+      end
+      rental_person = gets.chomp.to_i
+      puts 'Enter date'
```

Comment:
- [ ] [OPTIONAL] To improve the UX(User Experience ), you could add a date template/format when your print 'Enter date'. Such as DD/MM/YYYY.

Take a look at what you currently have:
<img width="231" alt="Screenshot 2023-01-05 at 15 55 36" src="https://user-images.githubusercontent.com/58692590/210811428-e9a3222d-4636-4372-9971-febc9f51dfc5.png">